.github/workflows/tests.yml

```diff
@@ -11,16 +11,17 @@ jobs:
     runs-on: ubuntu-latest
 
     strategy:
+      fail-fast: false
       matrix:
-        go: ['1.17', '1.16', '1.15', '1.14']
+        go: ['1.19']
```

Comment:
I'm all for using the latest and greatest Go release, but the PR description and the change here misses the rationale we need to go through for such changes.  The whole point of these list is that we _support_ those releases, and whether 1.14 is EOL by upstream or not, Ubuntu 18.04 is still a supported release, and ships Go 1.10 (!).

With that in mind, a few different points that we need to discuss for such changes:

1. What's indeed the oldest environment we care about compiling Pebble under?
2. The list above should contain every Go version since that requirement
3. The io.ReadAll function, provided as a motivation for this change, was just copied/moved out of ioutil.ReadAll

Reply:
`ioutil` is deprecated in Go 1.16+ so there's no good solution here. I would also like to use generics if possible in some work I'm currently doing.

> Ubuntu 18.04 is still a supported release, and ships Go 1.10 (!).

Sure, that doesn't stop them from updating to a later version of Go though?

I'm new to the project so I can't answer
> What's indeed the oldest environment we care about compiling Pebble under?

Juju is on Go 1.19 (even in 2.9) so we don't need an older version than that. I can't speak to anyone else using Pebble.

Reply:
Yeah, I suspect deprecated methods like `ioutil.ReadAll` will stick around for a looong time, given Go's conservative policy on no breaking changes. So that's not much of an incentive by itself to upgrade.

I don't know that we need to support the version of Go that ships with a particular Ubuntu release. That's more important for languages like Python which need the exact version of the interpreter, but Go compiles to stand-alone binaries, so you can easily build with Go (say) 1.19 and ship to a system which doesn't have Go at all.

If we need generics, it's obviously critical to have Go 1.18+ or they won't work. What's your use case for generics? (We haven't had a burning need for them to date.)

Either way, I'd be in favour of upgrading to Go 1.18 (I still think 1.19 is a bit premature!). But also, it'd be good to ask Cristovao and co on the ROCKS team what constraints they have. I've asked him [here](https://chat.canonical.com/canonical/pl/bmxgqxw4s7yfig1o3wrboafkra).

Reply:
Since ioutil is in the go1 tag (see https://github.com/golang/go/commit/b0683bd77a0779fd9a87a2a51bd9eb7969606ffe ) I would expect these methods to be deprecated until go2.

Reply:
Cristovao's reply about what they need for ROCKS:

> Ben: With using Pebble in ROCKS, what are your constraints w.r.t. the Go version Pebble is built with, if any? We're considering updating the minimum supported Go version to 1.18 or 1.19. See https://github.com/canonical/pebble/pull/171

> Cristovao: afaik, none. At the moment, we are installing Pebble from source, with Go 1.19 -> https://github.com/canonical/rockcraft/blob/21de155380e07772c35765cfa71e789b5ff8c5b6/rockcraft/project.py#L520
>
> for the future, we'll probably introduce a Pebble snap (wip [here](https://forum.snapcraft.io/t/classic-confinement-for-pebble/33059)), so the Go version becomes irrelevant for Rockcraft, as it would simply take a pre-compiled Pebble binary from its Snap